### PR TITLE
Fix the CI unit test and integration test failures

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/steps/utils.py [hunk 1]
```diff
@@ -150,6 +151,8 @@ def create_component_class(step: "BaseStep") -> Type[_SimpleComponent]:
     Returns:
         The component class.
     """
+    from tfx.dsl.component.experimental.decorators import _SimpleComponent
```
Comment (strickvl) [inline src/zenml/steps/utils.py:154]: For future curiosity: what is this and what does importing it fix? Looks like magic!

File: tests/conftest.py [hunk 2]
```diff
@@ -194,88 +194,88 @@ def clean_client(
 
 @pytest.fixture
 def sql_store() -> BaseZenStore:
-    with tempfile.TemporaryDirectory(suffix="_zenml_sql_test") as temp_dir:
-        store = SqlZenStore(
```
Comment (strickvl) [inline tests/conftest.py:198]: For my understanding, what was wrong with the context? It was deleting the temporary files while they were still needed?

**Review by strickvl — APPROVED** [review]
This looks good to me. Just had a couple of questions, which you're welcome to ignore :)

File: src/zenml/steps/utils.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/steps/utils.py:154]: This import drags along the whole tfx.experimental module, which loads a lot of other modules. Among them, cryptography, which, if loaded in memory, won't allow any package updates through `zenml integration install` on Windows.

The root problem is that running _any_ CLI command (`zenml integration *` included) imports all the modules from _all_ CLI commands in memory, which 1. takes time and 2. makes uninstalling/upgrading some packages with pip impossible on Windows.

File: tests/conftest.py (hunk 2 above)
Comment (fa9r) [reply tests/conftest.py:198]: Would be curious as well

Comment (stefannica) [reply tests/conftest.py:198]: the SQLite store database created in the temporary directory is still in use when the tests complete. Windows doesn't like if you're trying to remove open files. Without the context, the temp directory is left behind.

I vaguely remember you facing a similar issue with using this context approach in materializers.

**Review by fa9r — APPROVED** [review]
Thanks for fixing them!

**Pull request merged**